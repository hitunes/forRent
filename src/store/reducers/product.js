import * as ContactTypes from "../actions";
import { Products } from "../../data";

const initialState = {
  data: Products,
  filteredList: [],
  filter: "all",
};

export const ProductReducer = (state = initialState, action) => {
  switch (action.type) {
    case ContactTypes.HANDLE_SEARCH:
      return {
        ...state,
        filter: action.payload,
      };
    case ContactTypes.RANGE_FILTER:
      return {
        ...state,
      };
    case ContactTypes.CHECKBOX_FILTER:
      let filterOccurence = intersection(Products, action.payload);
      return {
        ...state,
        filteredList:
          filterOccurence.length === 0 ? state.data : filterOccurence,
      };
    case ContactTypes.TOGGLE_DETAILS:
      let data = [...state.data];
      data[findId(data, action.payload)].active = !data[
        findId(data, action.payload)
      ].active;
      return {
        ...state,
        data,
      };
    default:
      return state;
  }
};
function findId(array, id) {
  return array.findIndex(object => object.id === id);
}
const intersection = (a, b) => {
  const s = new Set(b);
  let c = a.filter(x => s.has(x.value) || s.has(x.type));
  return c;
};
