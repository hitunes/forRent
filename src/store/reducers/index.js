/*jshint esversion: 6 */
import { combineReducers } from "redux";
import { ProductReducer } from "./product";

export default combineReducers({
  product: ProductReducer,
});
