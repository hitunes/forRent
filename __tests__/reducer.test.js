import { ProductReducer } from "../src/store/reducers/product";

describe(" ProductReducer", () => {
  it("should return the initial state", () => {
    expect(ProductReducer(undefined, {})).toMatchSnapshot();
  });

  it("should handle HANDLE_SEARCH", () => {
    expect(
      ProductReducer(
        {
          filter: "all",
        },
        {
          type: "HANDLE_SEARCH",
        }
      )
    ).toMatchSnapshot();
  });
});
