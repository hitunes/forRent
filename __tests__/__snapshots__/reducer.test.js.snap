// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` ProductReducer should handle HANDLE_SEARCH 1`] = `
Object {
  "filter": undefined,
}
`;

exports[` ProductReducer should return the initial state 1`] = `
Object {
  "data": Array [
    Object {
      "addtionalInfo": Array [
        Object {
          "info": "6-speaker radio/CD system",
        },
        Object {
          "info": "Escaro black fabric",
        },
        Object {
          "info": "Hybrid System display",
        },
        Object {
          "info": "Vehicle Stability Control",
        },
        Object {
          "info": "Hill-start Assist Control",
        },
      ],
      "id": 1,
      "info": Array [
        Object {
          "icon": "usergroup-add",
          "type": "5 passengers",
        },
        Object {
          "icon": "folder",
          "type": "3 large suitcases, 2 small suitcase",
        },
        Object {
          "icon": "font-colors",
          "type": "automatic transmission",
        },
        Object {
          "icon": "fullscreen",
          "type": "air conditioning",
        },
        Object {
          "icon": "car",
          "type": "9 km/l",
        },
      ],
      "manufacturer": "Ford",
      "offer": "Unlimited free miles included",
      "price": 299,
      "productImage": "/images/car_3.png",
      "productName": "Ford Escape",
      "tag": Array [
        Object {
          "id": 1,
          "name": "top seller",
        },
      ],
      "type": "economy",
      "value": "ford",
    },
    Object {
      "addtionalInfo": Array [
        Object {
          "info": "6-speaker radio/CD system",
        },
        Object {
          "info": "Escaro black fabric",
        },
        Object {
          "info": "Hybrid System display",
        },
        Object {
          "info": "Vehicle Stability Control",
        },
        Object {
          "info": "Hill-start Assist Control",
        },
      ],
      "id": 2,
      "info": Array [
        Object {
          "icon": "usergroup-add",
          "type": "2 passengers",
        },
        Object {
          "icon": "folder",
          "type": "2 large suitcases, 2 small suitcase",
        },
        Object {
          "icon": "font-colors",
          "type": "automatic transmission",
        },
        Object {
          "icon": "fullscreen",
          "type": "air conditioning",
        },
        Object {
          "icon": "car",
          "type": "8 km/l",
        },
      ],
      "manufacturer": "Reno",
      "offer": "Unlimited free miles included",
      "price": 945.25,
      "productImage": "/images/car_1.png",
      "productName": "Reno",
      "tag": Array [],
      "type": "compact",
      "value": "reno",
    },
    Object {
      "addtionalInfo": Array [
        Object {
          "info": "6-speaker radio/CD system",
        },
        Object {
          "info": "Escaro black fabric",
        },
        Object {
          "info": "Hybrid System display",
        },
        Object {
          "info": "Vehicle Stability Control",
        },
        Object {
          "info": "Hill-start Assist Control",
        },
      ],
      "id": 8,
      "info": Array [
        Object {
          "icon": "usergroup-add",
          "type": "2 passengers",
        },
        Object {
          "icon": "folder",
          "type": "2 large suitcases, 2 small suitcase",
        },
        Object {
          "icon": "font-colors",
          "type": "automatic transmission",
        },
        Object {
          "icon": "fullscreen",
          "type": "air conditioning",
        },
        Object {
          "icon": "car",
          "type": "8 km/l",
        },
      ],
      "manufacturer": "Reno",
      "offer": "Unlimited free miles included",
      "price": 1445.25,
      "productImage": "/images/car_1.png",
      "productName": "Reno",
      "tag": Array [],
      "type": "standard",
      "value": "reno",
    },
    Object {
      "addtionalInfo": Array [
        Object {
          "info": "6-speaker radio/CD system",
        },
        Object {
          "info": "Escaro black fabric",
        },
        Object {
          "info": "Hybrid System display",
        },
        Object {
          "info": "Vehicle Stability Control",
        },
        Object {
          "info": "Hill-start Assist Control",
        },
      ],
      "id": 3,
      "info": Array [
        Object {
          "icon": "usergroup-add",
          "type": "5 passengers",
        },
        Object {
          "icon": "folder",
          "type": "1 large suitcases, 1 small suitcase",
        },
        Object {
          "icon": "font-colors",
          "type": "automatic transmission",
        },
        Object {
          "icon": "fullscreen",
          "type": "air conditioning",
        },
        Object {
          "icon": "car",
          "type": "13 km/l",
        },
      ],
      "manufacturer": "Chevrolet",
      "offer": "Unlimited free miles included",
      "price": 360.99,
      "productImage": "/images/car_2.png",
      "productName": "Chevy Aveo",
      "tag": Array [],
      "type": "compact",
      "value": "chevrolet",
    },
    Object {
      "addtionalInfo": Array [
        Object {
          "info": "6-speaker radio/CD system",
        },
        Object {
          "info": "Escaro black fabric",
        },
        Object {
          "info": "Hybrid System display",
        },
        Object {
          "info": "Vehicle Stability Control",
        },
        Object {
          "info": "Hill-start Assist Control",
        },
      ],
      "id": 4,
      "info": Array [
        Object {
          "icon": "usergroup-add",
          "type": "5 passengers",
        },
        Object {
          "icon": "folder",
          "type": "2 large suitcases, 1 small suitcase",
        },
        Object {
          "icon": "font-colors",
          "type": "automatic transmission",
        },
        Object {
          "icon": "fullscreen",
          "type": "air conditioning",
        },
        Object {
          "icon": "car",
          "type": "14 km/l",
        },
      ],
      "manufacturer": "Toyota",
      "offer": "Unlimited free miles included",
      "price": 845.25,
      "productImage": "/images/car_1.png",
      "productName": "Toyota Corolla",
      "tag": Array [],
      "type": "intermediate",
      "value": "toyota",
    },
    Object {
      "addtionalInfo": Array [
        Object {
          "info": "6-speaker radio/CD system",
        },
        Object {
          "info": "Escaro black fabric",
        },
        Object {
          "info": "Hybrid System display",
        },
        Object {
          "info": "Vehicle Stability Control",
        },
        Object {
          "info": "Hill-start Assist Control",
        },
      ],
      "id": 5,
      "info": Array [
        Object {
          "icon": "usergroup-add",
          "type": "5 passengers",
        },
        Object {
          "icon": "folder",
          "type": "2 large suitcases, 1 small suitcase",
        },
        Object {
          "icon": "font-colors",
          "type": "automatic transmission",
        },
        Object {
          "icon": "fullscreen",
          "type": "air conditioning",
        },
        Object {
          "icon": "car",
          "type": "14 km/l",
        },
      ],
      "manufacturer": "BMW",
      "offer": "Unlimited free miles included",
      "price": 1245.25,
      "productImage": "/images/bmw.jpeg",
      "productName": "BMW",
      "tag": Array [],
      "type": "standard",
      "value": "bmw",
    },
  ],
  "filter": "all",
  "filteredList": Array [],
}
`;
